scipy/stats/_sobol.pyx

```diff
@@ -232,40 +229,49 @@ cpdef void initialize_v(cnp.int_t[:, :] v, const int dim):
     # Multiply each column of v by power of 2:
     # v * [2^(maxbit-1), 2^(maxbit-2),..., 2, 1]
     pow2 = 1
-    for d in range(MAXBIT):
+    for d in range(maxbit):
         for i in range(dim):
-            v[i, MAXBIT - 1 - d] *= pow2
+            v[i, maxbit - 1 - d] *= pow2
         pow2 = pow2 << 1
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-cpdef void _draw(const int n,
-                 const int num_gen,
+cpdef void _draw(const cnp.uint64_t n,
+                 const cnp.uint64_t num_gen,
                  const int dim,
-                 cnp.int_t[:, :] sv,
-                 cnp.int_t[:] quasi,
-                 cnp.float_t[:, :] result) nogil:
-    cdef int i, j, l, qtmp
-    cdef int num_gen_loc = num_gen
+                 const int maxbit,
+                 cnp.uint64_t[:, :] sv,
+                 cnp.uint64_t[:] quasi,
+                 cnp.float_t[:, :] sample) nogil:
+    cdef int j, l
+    cdef cnp.uint64_t num_gen_loc = num_gen
+    cdef cnp.uint64_t i, qtmp
+
+    # largest possible integer
+    cdef cnp.uint64_t largest_number = 2 ** maxbit
```

Comment:
For the scale I am not sure because in his code, Sergei Kucherenko is just checking to not return more than `2**maxbit-1` points, but has `2**maxbit` for the scale. I will send him an email to check these.